include/mbedtls/check_config.h

```diff
@@ -320,11 +320,20 @@
 #endif
 
 #if defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED) &&                    \
-    ( !defined(MBEDTLS_ECJPAKE_C) || !defined(MBEDTLS_SHA256_C) ||      \
+    ( !defined(MBEDTLS_ECJPAKE_C) ||                                    \
       !defined(MBEDTLS_ECP_DP_SECP256R1_ENABLED) )
 #error "MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED defined, but not all prerequisites"
 #endif
 
+/* Use of EC J-PAKE in TLS requires SHA-256.
+ * This will be taken from MD is present, or from PSA if MD is absent.
+ * Note: ECJPAKE_C depends on MD_C || PSA_CRYPTO_C. */
+#if defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED) &&                    \
```

Comment:
You mention that its use in TLS requires `SHA-256` - is it about the ad-hoc KDF function, or is there another use too? If it's about the KDF, it would be better to state the requirement specifically: https://github.com/Mbed-TLS/mbedtls/pull/6115/files#diff-1844b20cf0e3b97c23e4dc032c1fd6cfe36e69b2a6ca27788eb49d3b17bb4663R95

Reply:
Currently use in TLS can only go through the legacy API (`mbedtls_ecjpake_xxx()` functions), where the custom KDF is implicit in `mbedtls_ecjpake_derive_secret()`, so I think the dependencies are correct for now, but indeed we'll need to update them again in #5886 when we start optionally using the PSA API, including the new KDF from #6115.